pkg/runtime/runtime.go

```diff
@@ -3476,10 +3491,18 @@ func (a *DaprRuntime) startReadingFromBindings() (err error) {
 	a.inputBindingsCtx, a.inputBindingsCancel = context.WithCancel(a.ctx)
 
 	for name, binding := range a.compStore.ListInputBindings() {
-		isSubscribed, err := a.isAppSubscribedToBinding(name)
-		if err != nil {
-			return err
+		var isSubscribed bool
+		m := binding.GetComponentMetadata()
+
+		if isBindingOfExplicitDirection(inputBinding, m) {
+			isSubscribed = true
```

Comment:
Is this a fair assumption to make? What if the user is using a Component YAML that is shared among multiple apps?

What I thought this PR was about, was skipping the OPTIONS request if an input binding is only used as output.

Reply:
Yeah, the whole point of the OPTIONS request to begin with was to test if a binding is an input by expecting the app to response successfully to the OPTIONS request. If the user explicitly sets the direction, the call is no longer needed..

Reply:
Ok. Do we also need the opposite? If the binding is marked as output explicitly, omit the OPTIONS request and say `isSubscribed=false`?

Reply:
> What I thought this PR was about, was skipping the OPTIONS request if an input binding is only used as output.

Yep that was achieved with https://github.com/dapr/dapr/pull/6509